qiskit_nature/operators/second_quantization/vibrational_op.py

```diff
@@ -0,0 +1,338 @@
+            for label, coeff in data
+        ):
+            raise TypeError("Data list must be [(str, number)].")
+
+        self._coeffs: np.ndarray
+        self._labels: List[str]
+
+        if isinstance(num_modals, int):
+            num_modals = [num_modals] * num_modes
+
+        self._num_modes = num_modes
+        self._num_modals = num_modals
+        self._register_length = sum(self._num_modals)
+
+        labels, coeffs = zip(*data)
+        self._coeffs = np.array(coeffs, np.complex128)
+
+        if not any('_' in label for label in labels):
+            # Dense label
+            if not all(len(label) == self._register_length for label in labels):
+                raise ValueError("Lengths of strings of label are different.")
+            label_pattern = re.compile(r"^[I\+\-NE]+$")
+            invalid_labels = [label for label in labels if not label_pattern.match(label)]
+            if invalid_labels:
+                raise ValueError(f"Invalid labels for dense labels are given: {invalid_labels}")
+            self._labels = list(labels)
+        else:
+            # Sparse label
+            _validate_vibrational_labels(data, num_modes, num_modals)
+            dense_labels = _convert_to_dense_labels(data, num_modes, num_modals)
+
+            ops: List[VibrationalOp] = []
+            for dense_label, coeff in dense_labels:
+                new_op = reduce(lambda a, b: a @ b, (
+                        VibrationalOp((label, 1), num_modes, num_modals) for label in dense_label
+                ))
+                # We ignore the type here because mypy only sees the complex coefficient
+                ops.append(coeff * new_op)  # type: ignore
+
+            op = reduce(lambda a, b: a + b, ops)
+
+            self._labels = op._labels.copy()
+
+    def __repr__(self) -> str:
+        if len(self) == 1:
+            if self._coeffs[0] == 1:
+                return f"VibrationalOp('{self._labels[0]}')"
+            return f"VibrationalOp({self.to_list()[0]})"
+        return f"VibrationalOp({self.to_list()})"  # TODO truncate
+
+    def __str__(self) -> str:
+        """Sets the representation of `self` in the console."""
+        if len(self) == 1:
+            label, coeff = self.to_list()[0]
+            return f"{label} * {coeff}"
+        return "  " + "\n+ ".join([f"{label} * {coeff}" for label, coeff in self.to_list()])
+
+    def __len__(self):
+        return len(self._labels)
+
+    @property
+    def num_modes(self) -> int:
+        """The number of modes.
+        Returns:
+            The number of modes
+        """
+        return self._num_modes
+
+    @property
+    def num_modals(self) -> List[int]:
+        """The number of modals.
+        Returns:
+            The number of modals
+        """
+        return self._num_modals
+
+    @property
+    def register_length(self) -> int:
+        """Getter for the length of the register that the VibrationalOp `self` acts on."""
```

Comment:
Just saying Gets the register length would suffice - the stuff about VibrationalOp self should be implicit right - these are all properties of the class.